test/driver.cpp

```diff
@@ -23,66 +23,82 @@
 
+using std::string;
+using std::string_view;
+using std::size_t;
+
 //must be defined in a unit-specific source file such as "array-test.cpp"
 void runTests();
 
-void processCmdLine(std::vector<std::string_view> arguments);
+void processCmdLine(char* arguments[], const size_t size);
+
+void replace_all(string& str, const string& substr, 
+   const string& toReplace);
 
 int main(int argc, char* argv[])
 {
-    std::vector<std::string_view> arguments;
-    for (int i = 0; i < argc; ++i)
-        arguments.emplace_back(argv[i]);
-    
-    processCmdLine(arguments);
+    processCmdLine(argv, argc);
 
    return getTestsFailed();
 }
 
-static const std::string defaultHeaderText("Running $exe");
-
-void processCmdLine(std::vector<std::string_view> arguments)
+void processCmdLine(char* arguments[], const size_t size)
 {
    bool printHeader{ true };
    bool printSummary{ true };
-   std::string headerText(defaultHeaderText);
-   std::string passReportMode("auto");
+   string headerText("Running $exe");
+   string_view passReportMode("auto");
    unsigned short failThreshold(0);
-   std::string_view fileOpenMode;
-   std::string outputFilename;
+   string_view fileOpenMode;
+   string outputFilename;
+
+   constexpr string_view option_name_header{ "-h" }, option_name_header_text{ "-ht" },
+      option_name_summary{ "-s" }, option_name_prm{ "-p" }, option_name_file{ "-f" },
+      option_name_file_overwrite{ "-fo" }, option_name_file_append{ "-fa" },
+      option_name_threshold{ "-t" };
+
+   constexpr string_view option_value_yes{ "yes" };
+   constexpr string_view option_value_detail{ "detail" },
+      option_value_indicate{ "indicate" }, option_value_none{ "none" },
+      option_value_auto{ "auto" };
 
-   // begin parsing cmd-line arguments
-   for (std::size_t i = 1; i < arguments.size(); i += 2)
+   //begin parsing cmd-line arguments
+   for (size_t i = 1; i < size; i += 2)
    {
-      if (arguments[i] == "-h")
-         printHeader = arguments[i + 1] != "no";
-      else if (arguments[i] == "-ht")
-         headerText = arguments[i + 1];
-      else if (arguments[i] == "-p")
-         passReportMode = arguments[i + 1];
-      else if (arguments[i] == "-s")
-         printSummary = arguments[i + 1] != "no";
-      else if (arguments[i] == "-t")
+      string_view option_name(arguments[i]), value(arguments[i + 1]);
+      
+      if (option_name == option_name_header)
+         printHeader = (value == option_value_yes);
+      else if (option_name == option_name_header_text)
+         headerText = value;
+      else if (option_name == option_name_prm)
+         passReportMode = value;
+      else if (option_name == option_name_summary)
+         printSummary = (value == option_value_yes);
+      else if (option_name == option_name_threshold)
       {
-         std::string_view& v = arguments[i + 1];
+         const string_view& v = arguments[i + 1];
          std::from_chars(v.data(), v.data() + v.size(), failThreshold);
```

Comment:
I believe there is no longer a need to use the temporary variable `v`:

```suggestion
         std::from_chars(value.data(), value.data() + value.size(), failThreshold);
```

Reply:
You are correct, there is no longer a need for variable `v`.